ntoskrnl/mm/ARM3/miarm.h

```diff
@@ -786,7 +786,8 @@ MI_MAKE_HARDWARE_PTE_KERNEL(IN PMMPTE NewPte,
 
     /* Check that we are not setting valid a page that should not be */
     ASSERT(ProtectionMask & MM_PROTECT_ACCESS);
-    ASSERT((ProtectionMask & MM_GUARDPAGE) == 0);
+    ASSERT(((ProtectionMask <= MM_WRITECOMBINE || ProtectionMask >= MM_OUTSWAPPED_KSTACK) &&
```

Comment:
Using `<=` and `>=` to test/compare flags/mask looks odd too:
add an explicit code comment?